test-suite/js/src/test/scala/org/scalajs/testsuite/library/WrappedMapTest.scala

```diff
@@ -0,0 +1,93 @@
+/*
+ * Scala.js (https://www.scala-js.org/)
+ *
+ * Copyright EPFL.
+ *
+ * Licensed under Apache License 2.0
+ * (https://www.apache.org/licenses/LICENSE-2.0).
+ *
+ * See the NOTICE file distributed with this work for
+ * additional information regarding copyright ownership.
+ */
+
+package org.scalajs.testsuite.library
+
+import org.junit.Assert._
+import org.junit.Assume.assumeTrue
+import org.junit.{ BeforeClass, Test }
+import org.scalajs.testsuite.utils.Platform
+
+import scala.collection.mutable
+import scala.reflect.ClassTag
+import scala.scalajs.js
+
+object WrappedMapTest {
+  @BeforeClass
+  def assumeRuntimeSupportsSet(): Unit = {
+    assumeTrue("Assume Map exists in Global",
+      js.typeOf(js.constructorOf[js.Map[_, _]]) != "undefined")
+  }
+}
+
+class WrappedMapTest {
+
+  // Methods we actually implement
+
+  @Test def testGet(): Unit = {
+    val map: mutable.Map[String, Any] =
+      js.Map("a" -> "a", "b" -> 6, "e" -> js.undefined)
+    assertTrue(map.get("a") == Some("a"))
+    assertTrue(map.get("b") == Some(6))
+    assertTrue(map.get("e") == Some(()))
+    assertTrue(map.get("f") == None)
+  }
+
+  @Test def testPlusAndMinusOperator(): Unit = {
+    val jsMap = js.Map[Int, String]()
+    val map: mutable.Map[Int, String] = jsMap
+
+    map += 1 -> "hello world"
+    assertEquals("hello world", jsMap(1))
+    map += 3 -> "foo bar"
+    assertEquals("foo bar", jsMap(3))
+    map -= 1
+    assertFalse(jsMap.get(1).isDefined)
+  }
+
+  @Test def testIterator(): Unit = {
+    val elems = ('a' to 'e').map(_.toString).zip(1 to 5)
+    val jsMap = js.Map[String, Int]()
+    val map: mutable.Map[String, Int] = jsMap
+
+    jsMap ++= elems
+
+    assertTrue(map.iterator.toList.sorted.sameElements(elems))
```

Comment:
The `.sorted` here should not be necessary. If it is actually necessary, then something must be wrong somewhere.